client/src/components/MusicPlayer/MusicPlayer.tsx

```diff
@@ -29,6 +38,67 @@ function MusicPlayer() {
     getUserInfo();
   }, [dispatch]);
 
+  // token persist
+  useEffect(() => {
+    if (sessToken) {
+      const queryString = window.location.search;
+      const urlParams = new URLSearchParams(queryString);
+      const hasError = urlParams.get('error');
+      const persistToken = localStorage.getItem('persist:token');
+      if (persistToken) {
+        const persistParse = JSON.parse(persistToken);
+
+        if (
+          hasError ||
+          hasTokenExpired() ||
+          persistParse.sessToken === 'undefined'
+        ) {
+          getRefreshToken();
+        }
+      }
+    }
+
+    // helper function to check if token expired
+    function hasTokenExpired() {
+      if (!sessToken || !timeStamp) {
+        return false;
+      }
+
+      const millisecondsElpased = Date.now() - Number(timeStamp);
+      return millisecondsElpased / 1000 > Number(tokenExpires);
+    }
+
+    // logout helper function
+    function logout() {
+      window.localStorage.removeItem('persist:token');
+      window.location.assign('/'); // not sure if this works, but trying to jus redirect to origin which should be the login page once localStorage clears
+    }
+
+    // getRefreshToken helper function -- logout if no refreshToken, otherwise fetch refreshToken
+    async function getRefreshToken() {
```

Comment:
the only one i couldnt move to a util file was the 'logout' function because it is currently using a dispatch. i'm currently trying to find a way to clear out local storage with redux-persist by using the dispatch, but for some reason the previous info comes back.

Reply:
Okay, I'm able to clear out local storage via redux persist now using the folllowing code:

`      persistor.pause();
      persistor.flush().then(() => {
        return persistor.purge();
      });`

source:
https://stackoverflow.com/questions/72265273/how-to-remove-value-from-localstorage-in-redux-persist